stripe-crypto-onramp/src/main/java/com/stripe/android/crypto/onramp/model/OnrampAuthenticateUserCallback.kt

```diff
@@ -0,0 +1,27 @@
+package com.stripe.android.crypto.onramp.model
+
+fun interface OnrampAuthenticateUserCallback {
+    fun onResult(result: OnrampAuthenticateUserResult)
+}
+
+/**
+ * Result of an OnRamp User Authentication operation.
+ */
+sealed interface OnrampAuthenticateUserResult {
+    /**
+     * The link authentication was successful.
+     * @param success A Boolean indicating if the authentication was successful.
+     *                If `false`, the operation was cancelled.
+     */
+    class Completed internal constructor(
+        val success: Boolean
+    ) : OnrampAuthenticateUserResult
+
+    /**
+     * Authentication failed.
+     * @param error The error that caused the failure
+     */
+    class Failed internal constructor(
+        val error: Throwable
+    ) : OnrampAuthenticateUserResult
+}
```

Comment:
can we make sure we match the definition in the API review? including the `Canceled` callback (so that integrators can differentiate between cancellations and failures). 

We can remove the `success` field, and later include the crypto `customerId`. 

<img width="723" height="507" alt="Image" src="https://github.com/user-attachments/assets/35edf757-fba5-46fd-bc12-7ab677600724" />